src/xmpp/xmpp-im/xmpp_client.h

```diff
@@ -60,12 +60,12 @@ namespace XMPP
         Q_OBJECT
 
     public:
-        Client(QObject *parent=0);
+        Client(QObject *parent = nullptr, EncryptionHandler *encryptionHandler = nullptr);
```

Comment:
I'd really prefer setMethod for the handler. This way you could change it for example on plugin enable/disable